src/functions/CreateComponentLoader.ts

```diff
@@ -0,0 +1,100 @@
+import { locationReplacement } from '#constants';
+import type { Config } from '#lib/types';
+import { findFilesRecursivelyRegex } from '@sapphire/node-utilities';
+import { Result } from '@sapphire/result';
+import { accessSync } from 'node:fs';
+import { mkdir, writeFile } from 'node:fs/promises';
+import { dirname, join, relative, resolve } from 'node:path';
+
+/**
+ * Regular expression pattern for matching files ending with JavaScript-like extensions.
+ * - `.js`
+ * - `.ts`
+ * - `.mjs`
+ * - `.cjs`
+ * - `.mts`
+ * - `.cts`
+ */
+const regexForFilesEndingWithJavaScriptLikeExtensions = /\.([mc])?[jt]s$/;
+
+/**
+ * Regular expression used to match loader files.
+ */
+const regexForLoaderFiles = new RegExp(`_load${regexForFilesEndingWithJavaScriptLikeExtensions.source}`);
+
+/**
+ * Generates loader file content
+ * @param dir The directory to generate the loader for
+ * @param targetDir The directory that the `_load.<ext>` file will be written to
+ */
+async function generateVirtualPieceLoader(dir: string, targetDir: string) {
+	console.log(`Generating virtual piece loader for ${targetDir}`);
+
+	const files: string[] = [];
+	for await (const file of findFilesRecursivelyRegex(targetDir, regexForFilesEndingWithJavaScriptLikeExtensions)) {
+		if (regexForLoaderFiles.test(file)) continue;
+		files.push(relative(targetDir, file).replace(regexForFilesEndingWithJavaScriptLikeExtensions, '.$1js'));
+	}
+
+	console.log(`Found ${files.length} ${dir} files`);
+
+	return `${files.map((file) => `import './${file}';`).join('\n')}\n`;
+}
+
+/**
+ * Creates component loaders
+ *
+ * We wrap the bulk of this function in a {@link Result.fromAsync} so that if any of the file writing fails
+ * that will bubble up as a failure overall.
+ *
+ * @param targetDir The directory that the `_load.<ext>` file will be written to
+ * @param config The config
+ * @returns Whether the loaders were created successfully
+ */
+export async function CreateComponentLoaders(targetDir: string, config: Config) {
+	const templateHeader = `// import this file in your entry point (index.${config.projectLanguage}) to load respective pieces`;
```

Comment:
Isn't this technically not needed for fs-based projects @kyranet? The file *will* get auto-imported, but nothing will be loaded from it 👀 

Also would be neat if we could make the entry point be taken from package.json but i will not block the PR on that

Reply:
The FS loader will load pieces twice if they're also loaded with `loadPiece`, first as virtual, then as non-virtual, that's why it should be disabled. That being said, `loadPiece` needs to be called for each piece, as such, this generator isn't fully automated.

The entrypoint bit is adding `import './listeners/_load.ts';` in the entrypoint file so the virtual files are registered, we can't automate this without using the TS API.